### ISSUE TITLE
Lint for use of format_args that could use format args capture

### ISSUE THREAD

**Issue by 5225225** [rust-lang/rust-clippy#8452]
### What it does

If the user writes some expression that uses format_args where any of the arguments could be captured, lint with a suggestion to have as many captures as possible.

This lint would have a suggestion category of MachineApplicable so it can be used to bulk move to using format args capture.

Not entirely sure which category this would belong in. I'm thinking either `restriction` (so you use this mostly as a refactoring tool, deny the lint and then `cargo clippy fix`), or `style` because it *is* a style matter. Depends how controversial the lint is.

I feel it would be *incredibly* noisy on any nontrivial crate, so making it warn by default might be a bad idea.

### Lint Name

uncaptured_format_args

### Category

style, restriction

### Advantage

Shorter, and the name of the argument is inline directly in the code.

### Drawbacks

If you have multiple arguments, and only some of them can be made inline, then the arguments will be out of order.

This could be mitigated against by only linting if we can move *all* of the arguments, with a configuration option to lint on *all* of them? That or 2 different lints.

### Example

```rust
let x = 1;
let y = format!("{}, {}", x, 10);
```

Could be written as:

```rust
let x = 1;
let y = format!("{x}, {}", 10);
```

**Comment by giraffate** [comment]
It seems to be a duplicate of #8368.

**Comment by 5225225** [comment]
Ah, you're right. Didn't find that in my searching, my bad!
